### PR TITLE
Homework14: Mariya Simonenko

### PR DESCRIPTION
**Pull request by Mary-si**

### CODE REVIEW

File: homework14_calculator.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+
+import ast
+
+
+def calculator(expression):
+    """калькулятор"""
+    try:
+        a = ast.parse(expression, mode="eval")
+        result = eval(compile(a, filename="", mode="eval"))
```
Comment (Stml89) [inline homework14_calculator.py:22]: if you are going with this, please add validation

File: homework14_file.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+# Напечатайте общее количество студентов,
+# количество студентов для каждой группы
+# и среднюю оценку для каждой группы. Допишите эту информацию в конец файла.
+# Передусмотрите возможные ошибки и обработайте их.
+
+
+class Students:
+    """студенты"""
+    count_students = 0
+    students_by_group: dict[str, list['Students']] = {}
```
Comment (Stml89) [inline homework14_file.py:17]: 1/ Use type hints everywhere or nowhere
2/ store and read all information from file

File: homework14_file.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        self.group = group
+        self.assessments = assessments
+        Students.count_students += 1
+        if group in Students.students_by_group:
+            Students.students_by_group[group].append(self)
+        else:
+            Students.students_by_group[group] = [self]
+
+    def __str__(self):
+        """общая информация о студентах"""
```
Comment (Stml89) [inline homework14_file.py:31]:
```suggestion
        """общая информация о студенте"""
```

File: homework14_file.py [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+
+student_1 = Students("Mariya Simonenko", "M-091", 3)
+student_2 = Students("Ivan Ivanov", "N-031", 4)
+student_3 = Students("Kseniya Petrova", "M-091", 5)
+
+try:
+    with open("students.txt", "w", encoding="utf-8") as f:
+        for student in [student_1, student_2, student_3]:
+            STUDENT_INFO = str(student)
```
Comment (Stml89) [inline homework14_file.py:60]: STUDENT_INFO in uppercase?
```suggestion
            STUDENT_INFO = str(student)
```

**Review by Stml89 — CHANGES_REQUESTED** [review]

File: homework14_calculator.py (hunk 1 above)
Comment (Mary-si) [reply homework14_calculator.py]: поправила

File: homework14_file.py (hunk 2 above)
Comment (Mary-si) [reply homework14_file.py:17]: поправила

File: homework14_file.py (hunk 3 above)
Comment (Mary-si) [reply homework14_file.py]: поправила

**Review by Stml89 — APPROVED** [review]

**Pull request merged**